### DISCUSSION TITLE
Moving from WCF to HTTPS

### DISCUSSION THREAD

**Discussion by kcabral817** [MarimerLLC/csla#2366] [Questions]
Does anyone have instructions, documents or examples on how to move from WCF endpoint to HTTPS endpoint.  I'm using WCF as a service as an end point.  I don't use anything outside of CSLA insofar as special WCF calls.  I do have a WPF front end and serialize across the network for the business objects.  I've never used kestrel or ASP so I'm really new at using that platform and hence the question. I do have all the books that Rocky has listed on his website for CSLA. Thanks for reading.

**Comment by rockfordlhotka** [comment]
There are some steps involved:

1. Make sure your business classes work with `MobileFormatter`
2. Set up a data portal controller on your server
3. Configure your clients to use `HttpProxy` instead of `WcfProxy`

The second two steps are pretty easy. The thing that _might_ be harder is if your business classes aren't ready for `MobileFormatter` yet.

## MobileFormatter

Microsoft is deprecating (and by .NET 6 pretty much not supporting) BinaryFormatter and NDCS - the two serializers used by legacy .NET Framework and WCF respectively. In 2008 when Silverlight came out, we wrote MobileFormatter as an alternative. MobileFormatter doesn't use reflection, which is nice. At the same time, to avoid reflection, business classes must _participate_ in serialization.

If you are using managed backing fields in your business classes, then CSLA does all this work for you. However, if you are using private backing fields, then you must implement code in your business classes to get/set those private field values during serialization.

In other words, if you are following the modern business class structure like you'll see in ProjectTracker and other samples, then your code will probably just work. If you are using older coding models then you'll have some work to do.

## Data Portal controller

Setting up an http-based data portal endpoint is as simple as creating a new API controller in your web server project. You can look at ProjectTracker to see how the [controller is implemented in the app server project](https://github.com/MarimerLLC/csla/blob/main/Samples/ProjectTracker/ProjectTracker.AppServerCore/Controllers/DataPortalController.cs). 

## Client configuration

There are many ways to configure a client, from app.config to the fluid API to dependency injection during startup. As we move toward .NET 6 and CSLA 6 the _preferred_ model is via DI during startup.

The ProjectTracker sample uses this technique. For example, you can see the [WPF client configuration](https://github.com/MarimerLLC/csla/blob/d6e6a4baddec8c4aa69b1bc5d9f94169af030633/Samples/ProjectTracker/ProjectTracker.Ui.WPF/App.xaml.cs#L19).
